rmgpy/molecule/group.py

```diff
@@ -1021,6 +1021,33 @@ def update(self):
         self.updateConnectivityValues()
         self.updateFingerprint()
 
+    def updateCharge(self):
+        """
+        Update the charge according to the valence_electron, total bond order, lone pairs
+        and radical electrons. When charged molecule is appearing in reactants, the product
+        template generated by applying recipe will result in wrong charge status. This
+        function is aimed to solve this problem
+        """
+        for atom in self.atoms:
+            if (len(atom.charge) == 1) & (len(atom.lonePairs) == 1) & (len(atom.radicalElectrons) == 1):
+                # if the charge of the group is not labeled, then no charge update will be
+                # performed. If there is a list of charge, no update either. Besides, this groupatom have to
+                # have enough information to be updated
+                atomtype = atom.atomType[0]
+                for element in allElements:
+                    if atomtype is atomTypes[element] or atomtype in atomTypes[element].specific:
+                        elementlabel = element
+                        bondOrder, lonePairs, radicalElectrons = 0, 0, 0
+                        import rmgpy.molecule.element as elements
+                        valence_electron = elements.PeriodicSystem.valence_electrons[elementlabel]
+                        for connectAtom, bond in atom.bonds.iteritems():
```

Comment:
If your not using one of the iterators in the loop (here `connectAtom`), it is a convention to name it with an underscore (`_`) so future programmers immediately understand it is not used on purpose, and won't think something is missing.